src/demo.rs

```diff
@@ -434,6 +437,77 @@ impl DemoManager {
     }
 }
 
+#[derive(Serialize, Deserialize)]
+pub struct VoteCastEventWrapped {
+    voter: Option<SteamID>,
+    voter_name: Option<String>,
+    choice: Option<String>,
+    event: VoteCastEvent,
+}
+
+impl VoteCastEventWrapped {
+    #[must_use]
+    pub fn from_vote_cast_event(event: VoteCastEvent, voter: Option<SteamID>) -> Self {
+        Self {
+            voter,
+            voter_name: None,
+            choice: None,
+            event,
+        }
+    }
+
+    pub fn set_choice(&mut self, choice: String) {
+        self.choice = Some(choice);
+    }
+
+    pub fn set_voter(&mut self, voter: Option<SteamID>) {
+        self.voter = voter;
+    }
+
+    pub fn set_vote_name(&mut self, name: String) {
+        self.voter_name = Some(name);
+    }
```

Comment:
Since it's just used in the broadcast handler thingy and doesn't really need any specific handling of it's internal fields, I'd probably just make the fields public and manipulate them directly instead of using setters.

Reply:
+1